src/mesh/amr_loadbalance.cpp

```diff
@@ -528,10 +528,62 @@ void Mesh::RedistributeAndRefineMeshBlocks(ParameterInput *pin, ApplicationInput
   Kokkos::Profiling::popRegion(); // Step 4
 
   MPI_Request *req_send, *req_recv;
-  // Step 5. allocate and start receiving buffers
-  Kokkos::Profiling::pushRegion("Step 5: Alloc buffer and start recv");
+
+  // Step 5. Allocate space for send and recieve buffers
+  Kokkos::Profiling::pushRegion("Step 5: Allocate send and recv buf");
+  size_t buf_size = 0;
   if (nrecv != 0) {
     recvbuf = new BufArray1D<Real>[nrecv];
+    for (int n = nbs; n <= nbe; n++) {
+      int on = newtoold[n];
+      LogicalLocation &oloc = loclist[on];
+      LogicalLocation &nloc = newloc[n];
+      if (oloc.level > nloc.level) { // f2c
+        for (int l = 0; l < nleaf; l++) {
+          if (ranklist[on + l] == Globals::my_rank) continue;
+          buf_size += bsf2c;
+        }
+      } else { // same level or c2f
+        if (ranklist[on] == Globals::my_rank) continue;
+        int size;
+        if (oloc.level == nloc.level) {
+          size = bssame;
+        } else {
+          size = bsc2f;
+        }
+        buf_size += size;
```

Comment:
Maybe make this a bit more succinct:
```suggestion
        buf_size += (oloc.level == nloc.level) ? bssame : bsc2f;
```